java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightEndpoint.java

```diff
@@ -39,6 +40,7 @@ public class FlightEndpoint {
   private final List<Location> locations;
   private final Ticket ticket;
   private final Instant expirationTime;
+  private final String appMetadata;
```

Comment:
Let's use `byte[]` since it can be binary. (ArrowBuf would be better but it needs explicit closing.)

Reply:
How should I handle a `byte[]` in the `toString` method?

Reply:
base64, or just omit it (or you could try to decode as a string and fall back to base64)

Reply:
`Arrays.toString(appMetadata)` seems like a good choice, but maybe a little odd if the metadata is an object converted to a bytearray.

Reply:
I'll go with base64 since that always works and it's easy enough to decode in case someone wants to see the stored value.